sdk/search/search-documents/src/generated/service/models/index.ts

```diff
@@ -761,8 +762,8 @@ export interface CharFilter {
 
 /** Base type for normalizers. */
 export interface LexicalNormalizer {
-  /** Identifies the concrete type of the normalizer. */
-  odatatype: string;
+  /** Polymorphic discriminator, which specifies the different types this object can be */
+  odatatype: "#Microsoft.Azure.Search.CustomNormalizer";
```

Comment:
I'm confused about this. Why would a generic `LexicalNormalizer` have the same `odatatype` as a `CustomNormalizer`?

Reply:
@xirzec  Yeah. It was little confusing to me too. Let me explain with an example. For the following swagger:

```json
"Similarity": {
  "discriminator": "@odata.type",
  .........
}

"ClassicSimilarity": {
  "x-ms-discriminator-value": "#Microsoft.Azure.Search.ClassicSimilarity",
  .........
}

"BM25Similarity": {
  "x-ms-discriminator-value": "#Microsoft.Azure.Search.BM25Similarity",
  .........
}
```

For the above swagger, the SDK will be generated as:

```js
export interface Similarity {
  odatatype: "#Microsoft.Azure.Search.ClassicSimilarity" | "#Microsoft.Azure.Search.BM25Similarity";
  .........
}

export type ClassicSimilarity = Similarity & {
  odatatype: "#Microsoft.Azure.Search.ClassicSimilarity";
  .........
};

export type BM25Similarity = Similarity & {
  odatatype: "#Microsoft.Azure.Search.BM25Similarity";
  .........
}
```

In the above code, look at the ```Similarity``` interface. It is the base class. But, the ```odatatype``` of ```Similarity``` is actually the union of ```odatatype``` of the two sub classes (```ClassicSimilarity``` & ```BM25Similarity```). 

Now, let us look into the ```LexicalNormalizer```. The swagger is:

```json
"LexicalNormalizer": {
  "discriminator": "@odata.type",
   ......
}

"CustomNormalizer": {
  "x-ms-discriminator-value": "#Microsoft.Azure.Search.CustomNormalizer",
   ..........
}
```

For the above swagger, the SDK looks like:

```js
export interface LexicalNormalizer {
  odatatype: "#Microsoft.Azure.Search.CustomNormalizer";
  .........
}

export type CustomNormalizer = LexicalNormalizer & {
  odatatype: "#Microsoft.Azure.Search.CustomNormalizer";
  .........
}
```

In the above code the base class ```LexicalNormalizer```, the odatatype is set to that of the ```CustomNormalizer``` which may look a little odd. 

But, consider there is another normalizer in the swagger such as this:

```json
"SampleNormalizer": {
  "x-ms-discriminator-value": "#Microsoft.Azure.Search.SampleNormalizer",
   ..........
}
```

then the resulting SDK will look like:

```js
export interface LexicalNormalizer {
  odatatype: "#Microsoft.Azure.Search.CustomNormalizer" | "#Microsoft.Azure.Search.SampleNormalizer";
  .........
}

export type CustomNormalizer = LexicalNormalizer & {
  odatatype: "#Microsoft.Azure.Search.CustomNormalizer";
  .........
}

export type SampleNormalizer = LexicalNormalizer & {
  odatatype: "#Microsoft.Azure.Search.SampleNormalizer";
  .........
}
```

The confusion is that we do not have more than one sub type of normalizer, it is looking a little odd. Am I making sense?

Reply:
Ohh! This does make sense and helps a lot. Thanks! 👍